internal/config/config.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type Config struct {
```

Comment:
Any specific reason you are marshaling this yourself now? Viper can do this and it will avoid a missed config setting like was done here with "factory" gettting missed and then overwritten without it.

Reply:
Viper was marshaling everything, even flags like `-v` or similar, and I didn't really like it. It might have been just me not knowing how to correctly use Viper though.

Reply:
I'm struggling with this thing also. I'm going to spend a little more time on it and see if i can clean it up. The "-v" one is the thing that's annoying. If you run a command with "-v" that happens to do a token refresh, then your config will get updated.